frontend/src/calldata/encoder.ts

```diff
@@ -0,0 +1,164 @@
+import type { CalldataEncodable } from './types';
```

Comment:
why is this whole module needed in the frontend? will it be part of GenLayerJS?

Reply:
yes, it will